app/backend/approaches/chatreadretrieveread.py

```diff
@@ -153,29 +155,39 @@ async def run_until_final_call(
             }
         ]
 
+        # Process results
+        current_model = self.hf_model if self.hf_model else self.chatgpt_model
+
+        # Load the Prompty object
+        query_prompty_path = SUPPORTED_MODELS.get(current_model) / "query.prompty"
+        query_prompty = Prompty.load(source=query_prompty_path)
+
         # STEP 1: Generate an optimized keyword search query based on the chat history and the last question
-        query_response_token_limit = 100
-        query_messages = build_messages(
-            model=self.chatgpt_model,
-            system_prompt=self.query_prompt_template,
-            tools=tools,
+        query_messages = query_prompty.render(
+            question=original_user_query,
             few_shots=self.query_prompt_few_shots,
             past_messages=messages[:-1],
-            new_user_content=user_query_request,
-            max_tokens=self.chatgpt_token_limit - query_response_token_limit,
         )
 
         chat_completion: Union[ChatCompletion, ChatCompletionOutput, AsyncIterable[ChatCompletionStreamOutput]] = (
             await self.llm_client.chat_completion(
-                messages=self.llm_client.format_message(query_messages),  # type: ignore
+                messages=ast.literal_eval(query_messages),  # type: ignore
                 # Azure OpenAI takes the deployment name as the model name
                 model=(
                     self.hf_model
                     if self.hf_model
                     else self.chatgpt_deployment if self.chatgpt_deployment else self.chatgpt_model
                 ),
-                temperature=0.0,  # Minimize creativity for search query generation
-                max_tokens=query_response_token_limit,  # Setting too low risks malformed JSON, setting too high may affect performance
+                temperature=(
+                    query_prompty._model.parameters["temperature"]
+                    if query_prompty._model.parameters["temperature"] is not None
+                    else 0.0
+                ),  # Minimize creativity for search query generation
+                max_tokens=(
+                    query_prompty._model.parameters["max_tokens"]
+                    if query_prompty._model.parameters["max_tokens"] is not None
+                    else 100
+                ),  # Setting too low risks malformed JSON, setting too high may affect performance
                 n=1,
                 tools=tools,
                 seed=seed,
```

Comment:
Maybe it wouldn't be a bad idea, as Prompty template can have unlimited LLM parameters that we might not even be aware off, to pass those params from the config as **kwargs, which should probably work.